src/array.cr

```diff
@@ -1201,6 +1201,13 @@ class Array(T)
     pop { raise IndexError.new }
   end
 
+  # Removes the last value from `self`.
+  # If the array is of size 0, the given block is called.
+  #
+  # ```
+  # a = [1]
+  # a.pop { "Testing" } #=> 1
+  # a.pop { "Testing" } #=> "Testing"
   def pop
```

Comment:
Is this method even meant to be publicly exposed? It sure is an odd duck in terms of behavior. If it's meant to be private, perhaps it's best if it's hidden from the [API docs](https://crystal-lang.org/api/0.20.1/Array.html#pop%28%26block%29-instance-method) altogether.